packages/backend/src/dbt/translator.ts

```diff
@@ -144,20 +145,21 @@ const convertTables = (allModels: DbtModelNode[]): Table[] => {
 }
 
 export const convertExplores = async (models: DbtModelNode[]): Promise<Explore[]> => {
-    const tables: Record<string, Table> = convertTables(models).reduce((prev, relation) => {
+    const allTables: Record<string, Table> = convertTables(models).reduce((prev, relation) => {
         return {...prev, [relation.name]: relation}
     }, {})
 
     const explores = models.map(model => {
         const tableNames = [model.name, ...(model.meta.joins || []).map(j => j.join)]
+        const tables = Object.fromEntries(tableNames.map(n => [n, allTables[n]]));
         return {
             name: model.name,
             baseTable: model.name,
             joinedTables: (model.meta.joins || []).map(join => ({
                 table: join.join,
-                sqlOn: join.sql_on,
+                sqlOn: renderExploreJoinSql(join.join, join.sql_on, {tables})
```

Comment:
This actually switches up the design quite a bit

Before the integrity of the Lightdash references weren't checked at compile time.

There's an outstanding ticket #44 to tackle this.

But just to note that rendering here is going to trigger any errors arising from an incorrect Lightdash config (which will happen commonly).

Let's not release after this PR and wait until #44 handles these errors gracefully. 

Then I like the idea that an `Explore` is all fully rendered, and an `Explore` cannot exist if the user has not specified it properly